packages/grafana-data/src/utils/dataLinks.ts

```diff
@@ -38,7 +38,11 @@ export type LinkToExploreOptions = {
 export function mapInternalLinkToExplore(options: LinkToExploreOptions): LinkModel<Field> {
   const { onClickFn, replaceVariables, link, scopedVars, range, field, internalLink } = options;
 
-  const interpolatedQuery = interpolateObject(link.internal?.query, scopedVars, replaceVariables);
+  const query =
+    typeof link.internal?.query === 'function'
+      ? link.internal.query({ replaceVariables, scopedVars })
```

Comment:
Do we need `replaceVariables/scopedVars` in the factory function? The query will be interpolated one line below in `interpolateObject`.

Reply:
The callback inspects scope vars and uses some more complex logic that goes beyond interpolation, see https://github.com/grafana/grafana/pull/98442/files#diff-ff5a9fcd5498eef32c39075e826af4e1a3e98c70f1a3e809f4f7ba2df7eaa58cR1161. 
Entirely different quries can be produced